internal/pkg/api/node/list.go

```diff
@@ -87,6 +89,23 @@ func NodeList(nodeGet *wwapiv1.GetNodeList) (nodeList wwapiv1.NodeList, err erro
 					fmt.Sprintf("%s:=:%s:=:%s:=:%s", n.Id.Print(), f.Field, f.Source, f.Value))
 			}
 		}
+	} else if nodeGet.Type == wwapiv1.GetNodeList_YAML || nodeGet.Type == wwapiv1.GetNodeList_JSON {
+		// handling the json and yaml format
+		filterNodes := node.FilterByName(nodes, nodeGet.Nodes)
+		nodeMap := make(map[string]node.NodeInfo)
+		for _, node := range filterNodes {
+			nodeMap[node.Id.Get()] = node
+		}
+
+		buf, _ := yaml.Marshal(nodeMap)
+
+		if nodeGet.Type == wwapiv1.GetNodeList_JSON {
+			jsonBuf, _ := yaml.YAMLToJSON(buf)
```

Comment:
Wouldn't this lead to the case that if json is selected the yaml is always added before the json?